src/webgpu/api/operation/render_pass/resolve.spec.ts

```diff
@@ -0,0 +1,204 @@
+        layout(location = 3) out vec4 fragColor3;
+        void main() {
+          fragColor0 = vec4(1.0, 1.0, 1.0, 1.0);
+          fragColor1 = vec4(1.0, 1.0, 1.0, 1.0);
+          fragColor2 = vec4(1.0, 1.0, 1.0, 1.0);
+          fragColor3 = vec4(1.0, 1.0, 1.0, 1.0);
+        }
+      `,
+    });
+
+    const pl = t.device.createPipelineLayout({ bindGroupLayouts: [] });
+    const colorStateDescriptors: GPUColorStateDescriptor[] = [];
+    for (let i = 0; i < t.params.numColorAttachments; i++) {
+      colorStateDescriptors.push({ format: kFormat });
+    }
+
+    const pipeline = t.device.createRenderPipeline({
+      vertexStage: { module: vertexModule, entryPoint: 'main' },
+      fragmentStage: { module: fragmentModule, entryPoint: 'main' },
+      layout: pl,
+      primitiveTopology: 'triangle-list',
+      rasterizationState: {
+        frontFace: 'ccw',
+      },
+      colorStates: colorStateDescriptors,
+      vertexState: {
+        indexFormat: 'uint32',
+        vertexBuffers: [],
+      },
+      sampleCount: 4,
+    });
+
+    const colorAttachments: GPUTexture[] = [];
+    const resolveTargets: GPUTexture[] = [];
+    const renderPassColorAttachmentDescriptors: GPURenderPassColorAttachmentDescriptor[] = [];
+
+    // The resolve target must be the same size as the color attachment. If we're resolving to mip
+    // level 1, the resolve target base mip level should be 2x the color attachment size.
+    const kResolveTargetSize: number = t.params.resolveTargetBaseMipLevel == 1 ? kSize * 2 : kSize;
+
+    let numResolveTargets: number = 0;
+
+    for (let i = 0; i < t.params.numColorAttachments; i++) {
+      colorAttachments.push(
+        t.device.createTexture({
+          format: kFormat,
+          size: { width: kSize, height: kSize, depth: 1 },
+          sampleCount: 4,
+          mipLevelCount: 1,
+          usage:
+            GPUTextureUsage.COPY_DST | GPUTextureUsage.COPY_SRC | GPUTextureUsage.OUTPUT_ATTACHMENT,
+        })
+      );
+
+      if (t.params.slotsToResolve.includes(i)) {
+        resolveTargets.push(
+          t.device.createTexture({
+            format: kFormat,
+            size: {
+              width: kResolveTargetSize,
+              height: kResolveTargetSize,
+              depth: t.params.resolveTargetBaseArrayLayer + 1,
+            },
+            sampleCount: 1,
+            mipLevelCount: t.params.resolveTargetBaseMipLevel + 1,
+            usage: GPUTextureUsage.COPY_SRC | GPUTextureUsage.OUTPUT_ATTACHMENT,
+          })
+        );
+
+        // Clear to black for the load operation. After the draw, the top left half of the attachment
+        // will be white and the bottom right half will be black.
+        renderPassColorAttachmentDescriptors.push({
+          attachment: colorAttachments[i].createView(),
+          loadValue: { r: 0.0, g: 0.0, b: 0.0, a: 0.0 },
+          storeOp: t.params.storeOperation,
+          resolveTarget: resolveTargets[numResolveTargets].createView({
```

Comment:
nit: slightly more clear we're using the resolve target we just made if we make make the texture and assign it, like `const resolveTarget = `, pass that resolveTarget here, and then push it into the list after. I don't think we need `numResolveTargets` as a variable because later, we can just iterate the whole list.

Reply:
Done.